### ISSUE TITLE
Watchdog caught collective operation timeout:

### ISSUE THREAD

**Issue by markAKB** [InternRobotics/InternNav#28]
您好请问遇到过如下的通信超时相关的报错吗？如果有的话，怎么解决的。我尝试了减少batchsize会有一定的缓解，但是最终还是会遇到。

[2025-08-18 18:02:05.261] [2025-08-18 18:02:05.261] {'loss': 0.0196, 'grad_norm': 0.2061067521572113, 'learning_rate': 5.000000000001034e-05, 'epoch': 2.476804883547509}
[2025-08-18 18:02:05.336] [2025-08-18 18:02:05.336] There are 6 training processes running
[2025-08-18 18:10:21.227] [2025-08-18 18:10:21.227] There are 6 training processes running
[2025-08-18 18:10:22.379] [2025-08-18 18:10:22.379] There are 6 training processes running
[2025-08-18 18:10:23.581] [2025-08-18 18:10:23.581] There are 6 training processes running
[2025-08-18 18:10:24.707] [2025-08-18 18:10:24.707] There are 6 training processes running
[2025-08-18 18:10:25.915] [2025-08-18 18:10:25.915] There are 6 training processes running
[2025-08-18 18:11:12.956] [2025-08-18 18:11:12.956] There are 6 training processes running
[2025-08-18 18:11:14.166] [2025-08-18 18:11:14.166] There are 6 training processes running
[2025-08-18 18:12:19.250] [2025-08-18 18:12:19.250] There are 6 training processes running
[rank4]:[E818 18:22:20.562780468 ProcessGroupNCCL.cpp:629] [Rank 4] Watchdog caught collective operation timeout: WorkNCCL(SeqNum=4170605, OpType=ALLREDUCE, NumelIn=592516, NumelOut=592516, Timeout(ms)=600000) ran for 600009 milliseconds before timing out.
[rank2]:[E818 18:22:20.624149113 ProcessGroupNCCL.cpp:629] [Rank 2] Watchdog caught collective operation timeout: WorkNCCL(SeqNum=4170605, OpType=ALLREDUCE, NumelIn=592516, NumelOut=592516, Timeout(ms)=600000) ran for 600016 milliseconds before timing out.
[rank1]:[E818 18:22:20.639285215 ProcessGroupNCCL.cpp:629] [Rank 1] Watchdog caught collective operation timeout: WorkNCCL(SeqNum=4170605, OpType=ALLREDUCE, NumelIn=592516, NumelOut=592516, Timeout(ms)=600000) ran for 600051 milliseconds before timing out.
[rank4]:[W818 18:22:21.332884140 Module.cpp:182] symbolizing C++ stack trace for exception; if this hangs, rerun with TORCH_DISABLE_ADDR2LINE=1...

[rank1]:[W818 18:22:21.332902536 Module.cpp:182] symbolizing C++ stack trace for exception; if this hangs, rerun with TORCH_DISABLE_ADDR2LINE=1...

[rank2]:[W818 18:22:21.332955721 Module.cpp:182] symbolizing C++ stack trace for exception; if this hangs, rerun with TORCH_DISABLE_ADDR2LINE=1...

[rank1]:[E818 18:22:22.495530836 ProcessGroupNCCL.cpp:2168] [PG ID 0 PG GUID 0(default_pg) Rank 1]  failure detected by watchdog at work sequence id: 4170605 PG status: last enqueued work: 4170633, last completed work: 4170604
[rank1]:[E818 18:22:22.495703893 ProcessGroupNCCL.cpp:667] Stack trace of the failed collective not found, potentially because FlightRecorder is disabled. You can enable it by setting TORCH_NCCL_TRACE_BUFFER_SIZE to a non-zero value.
[rank1]:[I818 18:22:22.495877570 NCCLUtils.hpp:372] Aborting ncclComm_ 0x55737dda1710 with reason: No abort reason provided.
[rank2]:[E818 18:22:22.504543522 ProcessGroupNCCL.cpp:2168] [PG ID 0 PG GUID 0(default_pg) Rank 2]  failure detected by watchdog at work sequence id: 4170605 PG status: last enqueued work: 4170633, last completed work: 4170604
[rank2]:[E818 18:22:22.504600003 ProcessGroupNCCL.cpp:667] Stack trace of the failed collective not found, potentially because FlightRecorder is disabled. You can enable it by setting TORCH_NCCL_TRACE_BUFFER_SIZE to a non-zero value.
[rank2]:[I818 18:22:22.504676106 NCCLUtils.hpp:372] Aborting ncclComm_ 0x5622757acc80 with reason: No abort reason provided.
[rank4]:[E818 18:22:22.505295091 ProcessGroupNCCL.cpp:2168] [PG ID 0 PG GUID 0(default_pg) Rank 4]  failure detected by watchdog at work sequence id: 4170605 PG status: last enqueued work: 4170633, last completed work: 4170604
[rank4]:[E818 18:22:22.505323627 ProcessGroupNCCL.cpp:667] Stack trace of the failed collective not found, potentially because FlightRecorder is disabled. You can enable it by setting TORCH_NCCL_TRACE_BUFFER_SIZE to a non-zero value.
[rank4]:[I818 18:22:22.505362703 NCCLUtils.hpp:372] Aborting ncclComm_ 0x563a973fa850 with reason: No abort reason provided.
honor119:93383:94491 [1] NCCL INFO [Service thread] Connection closed by localRank 1
honor119:93383:94339 [1] NCCL INFO comm 0x55737dda1710 rank 1 nranks 6 cudaDev 1 busId 66000 - Abort COMPLETE
[rank1]:[I818 18:22:55.234315935 NCCLUtils.hpp:372] Aborting ncclComm_ 0 with reason: No abort reason provided.
[rank1]:[E818 18:22:55.234345574 ProcessGroupNCCL.cpp:681] [Rank 1] Some NCCL operations have failed or timed out. Due to the asynchronous nature of CUDA kernels, subsequent GPU operations might run on corrupted/incomplete data.
[rank1]:[E818 18:22:55.234357994 ProcessGroupNCCL.cpp:695] [Rank 1] To avoid data inconsistency, we are taking the entire process down.
[rank1]:[E818 18:22:55.258140396 ProcessGroupNCCL.cpp:1895] [PG ID 0 PG GUID 0(default_pg) Rank 1] Process group watchdog thread terminated with exception: [Rank 1] Watchdog caught collective operation timeout: WorkNCCL(SeqNum=4170605, OpType=ALLREDUCE, NumelIn=592516, NumelOut=592516, Timeout(ms)=600000) ran for 600051 milliseconds before timing out.
Exception raised from checkTimeout at /pytorch/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:632 (most recent call first):
C++ CapturedTraceback:

**Comment by dedellzyw** [comment]
112核的cpu。目前bs是32，num_works是8，把num_works降为4？
